stable_baselines/ddpg/ddpg.py

```diff
@@ -634,9 +635,13 @@ def _store_transition(self, obs, action, reward, next_obs, done):
         :param reward: (float] the reward
         :param next_obs: ([float] or [int]) the current observation
         :param done: (bool) Whether the episode is over
+        :param info: (dict) extra values used to compute reward when using HER
         """
         reward *= self.reward_scale
-        self.replay_buffer.add(obs, action, reward, next_obs, float(done))
+        if isinstance(self.replay_buffer, HindsightExperienceReplayWrapper):
+            self.replay_buffer.add(obs, action, reward, next_obs, float(done), info)
+        else:
+            self.replay_buffer.add(obs, action, reward, next_obs, float(done))
```

Comment:
```suggestion
        # Store extra info that can be used to compute reward when using HER
        kwargs = {'info': info} if instance(self.replay_buffer, HindsightExperienceReplayWrapper) else {}
        self.replay_buffer.add(obs, action, reward, next_obs, float(done), **kwargs)
```

That's what I meant, it has the advantage of making clear what is different between the two cases.
if the line is too long, you can define an helper `is_using_her(self.replay_buffer)` in the off policy base class,
in fact that would be cleaner and avoid code duplication to do everything in the base class and call `self.add_to_buffer()`

Reply:
When I try to import `HindsightExperienceReplayWrapper` in `base_class.py` by doing `from stable_baselines.her import HindsightExperienceReplayWrapper` I get an error:

```
  File "main.py", line 4, in <module>
    from stable_baselines import HER, SAC, DQN, TD3, DDPG
  File "/home/tirafesi/Documents/stable-baselines/stable_baselines/__init__.py", line 1, in <module>
    from stable_baselines.a2c import A2C
  File "/home/tirafesi/Documents/stable-baselines/stable_baselines/a2c/__init__.py", line 1, in <module>
    from stable_baselines.a2c.a2c import A2C
  File "/home/tirafesi/Documents/stable-baselines/stable_baselines/a2c/a2c.py", line 7, in <module>
    from stable_baselines import logger
  File "/home/tirafesi/Documents/stable-baselines/stable_baselines/logger.py", line 17, in <module>
    from stable_baselines.common.misc_util import mpi_rank_or_zero
  File "/home/tirafesi/Documents/stable-baselines/stable_baselines/common/__init__.py", line 7, in <module>
    from stable_baselines.common.base_class import BaseRLModel, ActorCriticRLModel, OffPolicyRLModel, SetVerbosity, \
  File "/home/tirafesi/Documents/stable-baselines/stable_baselines/common/base_class.py", line 22, in <module>
    from stable_baselines.her import HindsightExperienceReplayWrapper
  File "/home/tirafesi/Documents/stable-baselines/stable_baselines/her/__init__.py", line 1, in <module>
    from stable_baselines.her.her import HER
  File "/home/tirafesi/Documents/stable-baselines/stable_baselines/her/her.py", line 3, in <module>
    from stable_baselines.common import BaseRLModel
ImportError: cannot import name 'BaseRLModel'
```

What am I doing wrong? I need this import or otherwise I can't check `isinstance(self.replay_buffer, HindsightExperienceReplayWrapper)`.

Reply:
You have a circular import issue... The trick is to do the import when the OffPolicy class is instantiated (see https://github.com/Stable-Baselines-Team/stable-baselines/blob/ed4e377a19ced11110d7c98782c119d1ff6ea380/stable_baselines/common/vec_env/__init__.py#L41 for an example)

Reply:
I think it should be OK now?

Reply:
Hm, it looks like type checking failed, but I'm not sure what I did wrong? Could you please take a look @araffin ?

Reply:
sorry, I will try to take a look this week, but I'm quite busy recently... so no promise